### DISCUSSION TITLE
Unable to download detailed records in "Drill to detail" and "Drill to Detail by"

### DISCUSSION THREAD

**Discussion by vaijayanthisundaram** [apache/superset#27607] [Ideas]
I am not be able to downloaded the detailed records in both "Drill to detail" and "Drill to Detail by" options. I can see the detailed records but see no option to download them.

Screenshot shows a bar graph created. Then I right click on a dimension and select "Drill to detail by " . I see the detailed records but no option to download them. Similarly for "Drill to detail" option.

browser type and version: Chrome Version 118.0.5993.117
superset version: superset version 3.0.1

<img width="1195" alt="Bar chart" src="https://github.com/apache/superset/assets/142204309/94d23ad8-b8d9-429d-8d24-f4e7c003a5da">

<img width="879" alt="Drill to detail" src="https://github.com/apache/superset/assets/142204309/51058c69-1273-4968-9a54-c7577885163d">

<img width="877" alt="Drill to details by" src="https://github.com/apache/superset/assets/142204309/cd3bca5e-1de8-47c3-9ce7-bff5e9d915e3">

**Comment by matiere57** [comment]
the same problem 
 - Superset ver. 3.1.1
 - Chrome ver. 123.0.6312.58

**Comment by rusackas** [comment]
Moving this to a discussion, since it's a feature request rather than a bug. We're open to PRs that enable this, if you want to tackle it. Personally, I think it's a great idea.

A potential related improvement might be leveraging the "Edit Chart" button - perhaps those filters and groupBys could be brought into the Chart Editor, so the chart reflects your drilling operations, and you could simply export from there.

<img width="872" alt="Pasted_Image_3_21_24__10_30 AM" src="https://github.com/apache/superset/assets/812905/82f3badc-0c66-4a60-9bf2-6c298e6bcad2">

**Comment by asimo101** [comment]
+1, this is a severely needed feature! 
But getting the user to edit the chart and then download is bad UX, especially if the user of the chart simply needs data to RCA, and is possibly a novice BI tool user. She wouldn't typically want to click on anything that says Edit.

In addition, when the underlying data layer is an OLAP cube which does not support a count(*) operation - as counts will work only on count-able metrics, and not at all on dimensions - the drill to detail today throws an error. I presume it is because the popup also shows the number of rows, and that query fails, for instance on cube.js. 

It would be helpful if there was a flag to disable getting row counts while doing drill-to-details.

**Comment by Erimus-Koo** [comment]
In the "edit chart" I do not find the method to export. There's a copy button, but all data is in one line.
![image](https://github.com/apache/superset/assets/17139670/c5bb635f-07b5-4612-a374-0f828da35f3e)

@rusackas If you design it as a new feature, just a little suggestion, a button in the drilled dialog may be a good idea. No jump out one more time.
e.g. 
![image](https://github.com/apache/superset/assets/17139670/bebe7b58-3f17-4ec8-b92c-c6cd997a8443)

**Comment by mlmaverick** [comment]
+1 badly need this feature.

**Comment by siddhant3030** [comment]
+1 Seriously I also need this please

**Comment by alexray92** [comment]
Really want this!

**Comment by datagene-github** [comment]
This Feature is needed

**Comment by kenho811** [comment]
Please add this feature. :pray: 

Please also add the ability to sort the columns. Otherwise, it does not provide much value when looking at data in a random order, without the ability to sort them

**Comment by cw1427** [comment]
Really need this feature

**Comment by afifdevtest** [comment]
+1 
If the feature is available on next release, it would be great and compete another open-source product 👍🏻

**Comment by fahadsamman** [comment]
+1

**Comment by smile2008** [comment]
greate feature
+1

**Comment by neuromantik33** [comment]
+1

**Comment by petrarca** [comment]
+1

**Comment by ramakinyanjui** [comment]
+1

**Comment by rusackas** [comment]
CC @geido @msyavuz 

Sounds like there's a ton of interest here, but no volunteers yet :D

**Comment by marianysilva** [comment]
+1

**Comment by guidefreitas** [comment]
+1

**Comment by rusackas** [comment]
Lots of +1s in the comment thread here, but make sure you upvote the top/opening comment, as that's how these are measured, rather than the conversational validation / thread-bumping. 

We already have code that lets you download CSV/XLS, so anyone is welcome to sort out how to wire that up here. Maybe a good Claude Code session if someone has time?

**Comment by Balaji-1905** [comment]
Can anyone be able to give a Tentative timeline for enabling the Download of the underlying data from a chart?

**Comment by ShivamYadav0** [comment]
Very Much needed feature!

**Comment by Igor-de-Sena** [comment]
+1

Does anyone have updates about it?

**Comment by mmbmurphy** [comment]
+1

**Comment by pva-kaizen** [comment]
I think I got it.
<img width="2536" height="1030" alt="downloadExcelSuperset" src="https://github.com/user-attachments/assets/b0db1f10-ca62-43d7-8e28-e2338c799a51" />

This download all rows in the dataset, at first it's downloading only the current page.

**Reply by Igor-de-Sena** [reply]
Is there a feature_flag that needs to be added to allow this?

**Reply by pva-kaizen** [reply]
I enabled that feature_flag however it does not show that download. So I created it on my own. 
It will be better if there's a feature flag that enables it automatically.

**Reply by raghuvarmacb** [reply]
How was the download feature implemented?

Was it done by modifying the Superset source code?

If yes, is this approach compatible with future Superset upgrades, or will it create maintenance issues?
